src/Components/pageComponents/auth/login/index.tsx

```diff
@@ -0,0 +1,312 @@
+                router.push('/userDashboard/tournament');
+                // router.push(configData.web.cominSoonUrl)
+            } else {
+                router.push('/spectatorDashboard');
+            }
+        } else {
+            router.push('/auth/401');
+        }
+    };
+
+    useEffect(() => {
+        const storedEmail = localStorage.getItem('email');
+        const storedPassword = localStorage.getItem('password');
+        if (storedEmail) {
+            setFieldValue('email', storedEmail);
+        }
+        if (storedPassword) {
+            setFieldValue('password', storedPassword);
+        }
+    }, [setFieldValue]);
+
+    // verify token
+    const handleVerifyToken = async (token: string) => {
+        setIsLoading(true);
+        try {
+            const verifyResponse = await sendRequest('/auth/verify', {
+                method: 'GET',
+                data: {
+                    token: token,
+                },
+            });
+
+            setIsLoading(false);
+
+            if (verifyResponse.status === 200) {
+                router.push('/adminDashboard/room');
+            } else {
+                setError('Google Sign-In failed');
+            }
+            if (verifyResponse.status === 200) {
+                router.push('/userDashboard');
+            } else {
+                setError('Google Sign-In failed');
+            }
+        } catch (error) {
+            setIsLoading(false);
+            setError('Google Sign-In failed');
+        }
+    };
+
+    const handleGoogleLogin = () => {
+        setIsLoading(true);
+
+        try {
+            window.location.href = 'http://localhost:5000/auth/google/callback';
+        } catch (error) {
+            setIsLoading(false);
+            setError('Google Sign-In failed');
+            console.error('Error during Google Sign-In:', error);
+        }
+    };
+
+    useEffect(() => {
+        if (typeof window !== 'undefined') {
+            const urlParams = new URLSearchParams(window.location.search);
+            const token = urlParams.get('token');
+            if (token) {
+                handleVerifyToken(token);
+            }
+        }
+    }, []);
+
+    // loader
+
+    const [isLoadingData, setLoadingData] = useState<boolean>(false);
+    const [errorData, showErrorData] = useState<string>('');
+
+    const handleVerifyTokenInLogin = async (token: string) => {
+        setLoadingData(true);
+        try {
+            const verifyResponse = await sendRequest(`auth/verify/?token=${token}`, {
+                method: 'GET',
+            });
+
+            setLoadingData(false);
+
+            if (verifyResponse.status === 200) {
+                router.push('/adminDashboard/room');
+            } else {
+                showErrorData('Google Sign-In failed');
+            }
+        } catch (errorData) {
+            setLoadingData(false);
+            showErrorData('Google Sign-In failed');
+        }
+    };
+
+    useEffect(() => {
```

Comment:
check this useEffect, remove unnecessary useEffects. group the similar ones

Reply:
okk sir